packages/wdio-browserstack-service/src/service.ts

```diff
@@ -148,6 +149,8 @@ export default class BrowserstackService implements Services.ServiceInstance {
                 if (this._observability) {
                     patchConsoleLogs()
 
+                    process.env.BROWSERSTACK_OBSERVABILITY_PRODUCT = this._isAppAutomate() ? 'app-automate' : (process.env.BROWSERSTACK_TURBOSCALE_INTERNAL === 'true' ? 'turboscale' : 'automate')
```

Comment:
Can we refactor this into a utility function that takes `WebdriverIO.Capabilities` and returns the value for the product? Writing into environment variables to pass along data across multiple classes is a not a good approach

Reply:
That was the intention however we don't make modifications to `WebdriverIO.Capabilities` when running sessions on the turboscale grid so we can't use those capabilities to fetch the product name.

Reply:
You can have 3 static utility methods, e.g.

```ts
export function isAppAutomate(caps: WebdriverIO.Capabilities): boolean {
    return !!caps['appium:app'] || !!caps['appium:options']?.app
}

export function isTurboScale(options: BrowserstackConfig): boolean {
    return Boolean(options.turboscale)
}

export function getProduct(options: BrowserstackConfig, caps: WebdriverIO.Capabilities): boolean {
    return isAppAutomate(caps)
        ? 'app-automate'
        : (isTurboScale(options) ? 'turboscale' : 'automate')
}
```

I strongly discouraged to just apply values to environment variables as you have no control as to if or when these values are assigned.

Reply:
I have made the code changes recommended above.